includes/sanitizers/class-amp-style-sanitizer.php

```diff
@@ -1987,7 +2005,18 @@ private function process_css_list( CSSList $css_list, $options ) {
 					$this->splice_imported_stylesheet( $css_item, $css_list, $options )
 				);
 			} elseif ( $css_item instanceof AtRuleSet ) {
-				if ( ! in_array( $css_item->atRuleName(), $options['allowed_at_rules'], true ) ) {
+				if ( in_array( $css_item->atRuleName(), $this->allowed_viewport_rules, true ) ) {
+					$output_format = new OutputFormat();
+					foreach ( $css_item->getRules() as $rule ) {
+						$rule_value = $rule->getValue();
+						if ( $rule_value instanceof Value ) {
+							$rule_value = $rule_value->render( $output_format );
+						}
+
+						self::$extracted_viewport_rules[] = [ $rule->getRule() => $rule_value ];
```

Comment:
Shouldn't we just store the properties in one array?

```suggestion
						self::$extracted_viewport_rules[ $rule->getRule() ] = $rule_value;
```

(Note the other discussion about changing `self::$extracted_viewport_rules` to be part of an array sent in the return value.)